src/wovnio/wovnphp/Url.php

```diff
@@ -251,7 +242,11 @@ public static function removeLangCode($uri, $lang_code, $store, $headers)
                 $customDomainLangs = $store->getCustomDomainLangs();
                 $default_lang = $settings['default_lang'];
                 $customDomainLangToRemove = $customDomainLangs->getCustomDomainLangByLang($lang_code);
-                $defaultCustomDomainLang = $customDomainLangs->getCustomDomainLangByLang($default_lang);
+                if ($customDomainLangToRemove->getSource()) {
```

Comment:
I would perhaps suggest changing initialization a bit and ensuring that `->getSource()` is never null and points to the final resolved value.

When initializing the custom domain langs, use the source override if it has one, otherwise give it the default lang source.

Reply:
I wish I've seen this comment earlier. Yep, doing that could save a lot of conditional checks.

Reply:
So like, we just have a fallback value for `source`? If you don't set one we set it to the `source` of default lang. And if default lang doesn't have a `source` then its `source` is itself.

Reply:
yep, kind of pre-compute the value.